### DISCUSSION TITLE
TW Icons: Replacement SETS And A Replacement Tool Seems Like A good Idea ...

### DISCUSSION THREAD

**Discussion by TiddlyTweeter** [morosanuae/tw-icons#4] [Ideas]
Say you import 50 icons to REPLACE (overwrite) 50 core icons...

Doing that manually one-by-one would be very laborious!

Could we think into HOW to make a simple tool to do that? ...

... For instance, a tool with the left column showing newly imported icons from TW Icons and, on the right, the core icons. 
You select one on left, then select one on right to replace. The new TW Icon overwrites the core version.

You get the idea? Ultimately we could maybe simply have SETS that overwrite all core icons for visual consistency?

Thoughts?

**Comment by morosanuae** [comment]
@TiddlyTweeter I need to better understand what you're trying to accomplish. What are the reasons behind overwriting the core icons? Can you give me an example?

**Reply by TiddlyTweeter** [reply]
Right. Understood! The reasons should be clearer once you can see some kind of example.

It will take me some days to make a decent illustration as I'm not a programmer :-).

**Comment by morosanuae** [comment]
An illustration would be good but it takes time and effort. I was just thinking there is a simple (and obvious) reason for overwriting (replacing) the core icons.

**Reply by TiddlyTweeter** [reply]
@morosanuae, reasons on SET replacement are, I think ...

- IF you use icons from ONE library on TW Icons there is likely visual SIZE (especially  padding) consistency. So replacing would likely remove issues to do with "baseline" differences between new and core as the core has been replaced by a complete set of new icons. That avoids having to individually hack icons or have to do CSS jujitsu one-by-one.

- One library style across all icons would likely achieve a better VISUAL consistency. Different icon sets at first may look compatible, but I think, in practice, if you mix icons from different libraries in one project it could become a headache. IMO it is better to stick with one library for one application.

My current quick 2 cents. I might be wrong. TT

**Reply by TiddlyTweeter** [reply]
> An illustration would be good but it takes time and effort. I was just thinking there is a simple (and obvious) reason for overwriting (replacing) the core icons.

Ciao @morosanuae. UPDATE. 

I have looked much more closely at the libraries. It is clear NOW to me that _a tool to auto-update core icons would NOT work_ reliably.

WHY? Because individual libraries implement styling, particularly of "stroke" v. "fill" in some DIFFERENT ways. SOME of which cannot be altered by external CSS without editing the SVG.

What that means in practice is that the user, to "conform" libraries "visually" to the standard TW UI will _sometimes_ need to edit `<path` embedded styling (e.g. for "stroke" and "fill") manually in the icon.

ALSO **(brilliance by YOU! :-)** every icon has a UNIQUE class. So, that too would also need to be manually edited IF you replaced a core icon so that  replacement SVG got the ORIGINAL class of the core icon restored.

_Hope this is clear!_

But, basically, I think we rather need to help users understand a bit better what SVG actually is and does FOR our UI purposes in TW?

Thoughts
TT
